packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -6489,17 +6489,28 @@ class Flow extends MultiChildRenderObjectWidget {
 ///
 /// {@tool snippet}
 ///
-/// This sample demonstrates how to assign a [SelectionRegistrar] for RichTexts
-/// in the SelectionArea subtree.
+/// This sample demonstrates how to assign a [SelectionRegistrar] for [RichText]s
+/// in the [SelectionArea] subtree.
 ///
 /// ![](https://flutter.github.io/assets-for-api-docs/assets/widgets/rich_text.png)
 ///
 /// ```dart
-/// RichText(
-///   text: const TextSpan(text: 'Hello'),
-///   selectionRegistrar: SelectionContainer.maybeOf(context),
-///   selectionColor: const Color(0xAF6694e8),
-/// )
+/// Widget build(BuildContext context) {
+///   return SelectionArea(
+///     // SelectionArea adds a SelectionRegistrar inherited widget
+///     // to its subtree through SelectionContainer. A new BuildContext
+///     // is necessary to access this inherited widget.
+///     child: Builder(
+///       builder: (BuildContext innerContext) {
+///         return RichText(
+///           text: const TextSpan(text: 'Hello'),
+///           selectionRegistrar: SelectionContainer.maybeOf(innerContext),
```

Comment:
What do you think of adding a `SelectionContainer.of` method and using that here? We fully expect there to be a `SelectionRegistrar` in `innerContext` here.

If the user copies this code but removes the `Builder` widget, it'll seem like the `RichText` selection is broken, when instead you're using inherited widgets wrong. If we use something like `SelectionContainer.of` here, it'd be clearer to the user that the problem is a misuse of inherited widgets.

Reply:
That sounds like a good idea to me! Instead of silently not working users are given an error, thanks for the suggestion.

Reply:
Looks great, thanks!